classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -488,4 +495,74 @@ protected function setFallbackVersion(): ?string
         }
         return null;
     }
+
+    /**
+     * Check the contexts' contact details before upgrade
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function checkContactSetting(): void
+    {
+        $primaryLocale = DB::table('site')
+            ->select(['primary_locale'])
+            ->first()
+            ->primary_locale;
+
+        // Get all the enabled context with context name/title
+        $contexts = DB::table("{$this->getContextTable()} AS context")
+            ->select([
+                "context.{$this->getContextKeyField()}", 
+                "context.enabled", 
+                "context_setting.locale",
+                "context_setting.setting_name",
+                "context_setting.setting_value"
+            ])
+            ->join("{$this->getContextSettingsTable()} AS context_setting", fn($join) =>
+                $join
+                    ->on("context.{$this->getContextKeyField()}", "=", "context_setting.{$this->getContextKeyField()}")
+                    ->where("context_setting.locale", "=", $primaryLocale)
+                    ->where("context_setting.setting_name", "=", "name")
+            )
+            ->where("context.enabled", 1)
+            ->get();
+        
+        // Get all the context IDs for which contact name/email is set
+        $contextIdsWithValidContact = DB::table("{$this->getContextTable()} AS context")
```

Comment:
For brevity, I'd use a subquery instead of joins here and just check if those settings exist, something like:
```sql
SELECT `journal_id` FROM `journals` WHERE `journal_id` IN (SELECT `journal_id` FROM `journal_settings` WHERE `setting_name` IN ('contactEmail', 'contactName'))
...
```

Reply:
Right, SubQuery is a better choice and through that I can do it in one single query rather than multiple query and diffing . Updated .